data_steward/analytics/cdr_ops/incremental_load_qc.py

```diff
@@ -0,0 +1,348 @@
+    run_as, target_scopes=IMPERSONATION_SCOPES)
+client = BigQueryClient(project_id, credentials=impersonation_creds)
+pid_or_rid = 'research_id' if is_deidentified == 'True' else 'person_id'
+
+# ## QC 1. (Only for `includes_aian == False`) Confirm no AIAN participants data is included in the new dataset
+# `incremental_dataset` contains AIAN participants' records. <br>We must exclude those in `new_dataset`
+# when creating a WITHOUT AIAN dataset.
+
+# +
+if includes_aian == "True":
+    success_msg = "Skipping this check because the new dataset can include AIAN participants.<br><br>"
+    render_message('', success_msg=success_msg)
+
+else:
+    person_id_tables_query = JINJA_ENV.from_string('''
+        SELECT table_name
+        FROM `{{project}}.{{new_dataset}}.INFORMATION_SCHEMA.COLUMNS`
+        WHERE column_name = "person_id"
+        ORDER BY table_name
+    ''').render(project=project_id, new_dataset=new_dataset)
+    pid_table_list = client.query(person_id_tables_query).to_dataframe().get(
+        'table_name').to_list()
+
+    query = JINJA_ENV.from_string('''
+        SELECT
+            \'{{table}}\' as table_name,
+            COUNT(*) AS aian_row_count
+        FROM `{{project}}.{{new_dataset}}.{{table}}`
+        WHERE person_id IN (
+            SELECT {{pid_or_rid}} FROM `{{project}}.{{aian_lookup_dataset}}.{{aian_lookup_table}}`
+        )
+        HAVING COUNT(*) > 0
+    ''')
+    queries = [
+        query.render(project=project_id,
+                     new_dataset=new_dataset,
+                     table=table,
+                     aian_lookup_dataset=aian_lookup_dataset,
+                     aian_lookup_table=aian_lookup_table,
+                     pid_or_rid=pid_or_rid) for table in pid_table_list
+    ]
+    union_all_query = '\nUNION ALL\n'.join(queries)
+
+    df = execute(client, union_all_query)
+
+    success_null_check = f"No AIAN participants' records are found in {new_dataset}.<br><br>"
+    failure_null_check = (
+        "There are <b>{count}</b> tables that might have "
+        "AIAN participants' records. <br>Look at these tables and investigate why AIAN "
+        "records are there: <b>{tables}</b><br><br>")
+
+    render_message(df,
+                   success_null_check,
+                   failure_null_check,
+                   failure_msg_args={
+                       'count': len(df),
+                       'tables': ', '.join(df.table_name)
+                   })
+# -
+
+# ## QC 2. Confirm there are no duplicate `OBSERVATION_ID`s
+# Any observation records from `incremental_dataset` got new `OBSERVATION_ID`s
+# assigned in `new_dataset` to avoid ID duplicates. <br>We must confirm all the
+# `OBSERVATION_ID`s in `new_dataset` are unique. <br>The same goes to
+# `_mapping_observation` and `obsesrvation_ext`.
+
+# +
+obs_tables = ['observation', 'observation_ext'
+             ] if is_deidentified == 'True' else [
+                 'observation', '_mapping_observation', 'observation_ext'
```

Comment:
FYI:  This raised an exception for us  on a test run.  With `is_deidentified = 'False'` we ran this against some combined datasets just to check query assumptions.  This failed on just a combined dataset because `observation_ext` did not exist.  It will exist in a combined release dataset but will not exist in an rdr or combined backup dataset.

Reply:
I updated the logic to check if it's `combined_release`, `deid`, or else. I also updated the main remediation script to have the same `if` logic:
https://github.com/all-of-us/curation/pull/1489/commits/7f18d4c7ca56c7dc1072a96dc600777aecc6b091#diff-638190b97f9a117d7f9c2a0fc16996cb1608d05f353b383919366a8b7aec007f